ts/client/scripts/risk.ts

```diff
@@ -0,0 +1,574 @@
+    // TODO USDC
+    const bank = banks[0];
+
+    const onChainPrice = (
+      await (
+        await fetch(`https://price.jup.ag/v4/price?ids=${bank.mint}`)
+      ).json()
+    )['data'][bank.mint.toBase58()]['price'];
+
+    if (bank.reduceOnly > 0 || bank.tokenIndex === usdcBank.tokenIndex) {
+      continue;
+    }
+
+    // Sum of all liabs, these liabs would be acquired by liqor,
+    // who would immediately want to reduce them to 0
+    // Assuming liabs need to be bought using USDC
+    const liabs =
+      // Max liab of a particular token that would be liquidated to bring health above 0
+      mangoAccountsWithHealth.reduce((sum, a) => {
+        // How much would health increase for every unit liab moved to liqor
+        // liabprice * (liabweight - (1+fee)*assetweight)
+        const tokenLiabHealthContrib = bank.price.mul(
+          bank.initLiabWeight.sub(
+            ONE_I80F48().add(bank.liquidationFee).mul(usdcBank.initAssetWeight),
+          ),
+        );
+        // Abs liab/borrow
+        const maxTokenLiab = a.account
+          .getTokenBalance(bank)
+          .min(ZERO_I80F48())
+          .abs();
+        // Health under 0
+        const maxLiab = a.health
+          .min(ZERO_I80F48())
+          .abs()
+          .div(tokenLiabHealthContrib)
+          .min(maxTokenLiab);
+
+        // DEBUG
+        // if (
+        //   a.health.toNumber() < -10000000 ||
+        //   maxLiab.gt(I80F48.fromNumber(10000))
+        // )
+        //   console.log(
+        //     `https://app.mango.markets/?address=${
+        //       a.account.publicKey
+        //     }, health ${a.health.toNumber().toLocaleString()}, bank ${
+        //       bank.name
+        //     }, maxLiab ${maxLiab.toNumber().toLocaleString()}`,
+        //   );
+
+        return sum.add(maxLiab);
+      }, ZERO_I80F48());
+    const liabsInUsdc =
+      // convert to usdc, this is an approximation
+      liabs
+        .mul(bank.price)
+        .floor()
+        // jup oddity
+        .min(I80F48.fromNumber(99999999999));
+    const pi1 = !liabsInUsdc.eq(ZERO_I80F48())
+      ? await computePriceImpactOnJup(
+          liabsInUsdc.toString(),
+          USDC_MINT,
+          bank.mint.toBase58(),
+        )
+      : { priceImpactPct: 0, outAmount: 0 };
+
+    // Sum of all assets which would be acquired in exchange for also acquiring
+    // liabs by the liqor, who would immediately want to reduce to 0
+    // Assuming assets need to be sold to USDC
+    const assets = mangoAccountsWithHealth.reduce((sum, a) => {
+      // How much would health increase for every unit liab moved to liqor
+      // assetprice * (liabweight/(1+liabliqfee) - assetweight)
+      const tokenAssetHealthContrib = bank.price.mul(
+        Array.from(group.banksMapByTokenIndex.values())
+          .flat()
+          .map((bank) => bank.initLiabWeight)
+          .reduce((prev, curr) => (prev.lt(curr) ? prev : curr))
```

Comment:
This would be easier for me to read if the min initLiabWeight was computed separately beforehand.